hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1255,10 +1255,14 @@ protected void setWriteSchemaForDeletes(HoodieTableMetaClient metaClient) {
           throw new HoodieIOException("Latest commit does not have any schema in commit metadata");
         }
       } else {
+        String schemaStr = new TableSchemaResolver(metaClient).getTableAvroSchema().toString();
```

Comment:
The whole method can be replaced by the new logic with try-catch block:
```
String schemaStr = new TableSchemaResolver(metaClient).getTableAvroSchema().toString();
config.setSchema(schemaStr);
```

Reply:
address it.